docs/guides/authjs-migration.mdx

```diff
@@ -0,0 +1,373 @@
+
+Use [`auth()`](/docs/references/nextjs/auth#auth) if you are using App Router or [`getAuth()`](/docs/references/nextjs/get-auth#get-auth) if you are using Pages Router to read your user data.
+
+<CodeBlockTabs type="installer" options={["App Router", "Pages Router"]}>
+
+```tsx failename="app/page.tsx"
+import { auth, currentUser } from "@clerk/nextjs";
+ 
+export default async function Page() {
+ 
+  const { userId } = auth();
+  console.log(userId)
+
+  return (
+    <p>Home Page</p>
+  )
+
+}
+```
+
+```tsx filename="app/page.tsx"
+
+export async function getServerSideProps(context) {
+  const session = getAuth(context.req);
+	
+	return { props: { ...buildClerkProps(ctx.req) } };
+};
+```
+</CodeBlockTabs>
+
+#### Client Side
+
+Replace Auth's `useSession` with Clerk's hooks.
+
+```tsx filename="app/page.tsx"
+'use client'
+import { useSession } from "next-auth/react"
+
+export default function Home() {
+  const { data: session } = useSession()
+  console.log(session)
+
+  return (
+    <p>Home Page</p>
+  )
+}
+```
+
+Use [`useAuth()`](/docs/references/react/use-auth) if you need basic authentication information or [`useUser()`](/docs/references/react/use-user) if you need the full user object.
+
+```tsx filename="app/page.tsx"
+'use client'
+import { useAuth, useUser } from "@clerk/nextjs";
+
+export default function Home() {
+  const { userId, sessionId } = useAuth();
+  const { isSignedIn, user } = useUser();
+  console.log(userId, sessionId, isSignedIn, user)
+
+  return (
+    <p>Home Page</p>
+  )
+}
+```
+
+### User IDs as Foreign Keys  
+
+When you migrate to Clerk, you will likely need to resolve the foreign key that you used in your database. If you used the `userId` from NextAuth.js, you could resolve this issue with one of the two following options.
+
+#### Use Clerk's `externalId` field
+
+When you migrate user data from Auth's database to Clerk, Clerk generates new user IDs for each user. If you are using existing user IDs as foreign keys in your database (e.g. in a `user_id` column), you can save those IDs as the user's `externalId` in Clerk. This `externalId` can be included in the session token by adding the following [customization](/docs/backend-requests/making/custom-session-token). The following example will return the `externalId` if one is present, and if not will return the Clerk `userId`.
+
+```json
+{
+	"userId": "{{user.external_id || user.id}}"
+}
+``` 
+
+In your application, you can access this combined `userId` via session claims.
+
+```tsx filename="app/page.tsx"
+import { auth } from "@clerk/nextjs";
+ 
+export default async function Page() {
+ 
+  const { sessionClaims: { userId } } = auth()
+  console.log(userId)
+
+  return (
+    <p>Home Page</p>
+  )
+
+}
+```
+
+<Callout type="info">
+You can not access the above from the client side. If you are using one of Clerk's hooks then you will need to check if `externalID` exists and if not then read the `userId`. 
+</Callout>
+
+#### Update your database
```

Comment:
```suggestion
#### Update your Auth.js database 
```

Reply:
The database isn't really Auth.js's or Clerk's, but the application. Calling it `Auth.js database` could be confusing and isn't accurate.